### PR TITLE
Add Pre Preview Readme

### PR DESCRIPTION
**Pull request by nplasterer**
This adds a more detailed Readme for devs who want to experiment with the react native SDK while it is in development.

### CODE REVIEW

**Comment by nplasterer** [comment]
I'm not sure how we can make it really really clear that only Android is working and not iOS? Any thoughts @jhaaaa Or do you think the way it is written is clear enough 🤷‍♀️

File: README.md [hunk 1]
```diff
@@ -1,21 +1,212 @@
...
+`xmtp-react-native` provides a React Native implementation of an XMTP message API client for use with Android and iOS apps.
 
-```sh
+Use `xmtp-react-native` to build with XMTP to send messages between blockchain accounts, including DMs, notifications, announcements, and more.
+
+This SDK is in **Pre Preview** status and ready for you to start experiment with.
+
+However, we do **not** recommend using Pre Preview software in production apps. Software in this status going to change as we iterate on features and feedback.
+
+Specifically, this SDK is working on iOS support and building out more functionality across the native bridges.
```
Comment (dmccartney) [inline README.md]: ~iOS~ Android?
Or maybe:
> This SDK only supports basic Android text messaging. We're still working to support iOS and to support content types beyond text.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,212 @@
...
+
+To keep up with the latest SDK developments, see the [Issues tab](https://github.com/xmtp/xmtp-react-native/issues) in this repo.
+
+To learn more about XMTP and get answers to frequently asked questions, see [FAQ about XMTP](https://xmtp.org/docs/dev-concepts/faq).
+
+![x-red-sm](https://user-images.githubusercontent.com/510695/163488403-1fb37e86-c673-4b48-954e-8460ae4d4b05.png)
+
+## Example apps
+
+For a basic demonstration of the core concepts and capabilities of the `xmtp-react-native` client SDK, see the Android and iOS [example apps](https://github.com/xmtp/xmtp-react-native/tree/main/example).
```
Comment (dmccartney) [inline README.md]: ~see the Android and iOS [example apps]~
maybe just:
> ... see the [example app].

File: README.md [hunk 3]
```diff
@@ -1,21 +1,212 @@
...
+
+You can run the example app by:
+
+```bash
+cd example
+yarn install
+yarn start
+a #for Android
+i #for iOS
+```
```
Comment (dmccartney) [inline README.md]: Better, imo:
```bash
$ yarn
$ yarn example start
$ yarn example android
```
Or just point them to `CONTRIBUTING.md` (which explains that more completely)

File: README.md [hunk 4]
```diff
@@ -1,21 +1,212 @@
...
+To accomplish this, you can pass a context with a `conversationId` when you are creating a conversation. We recommend conversation IDs start with a domain, to help avoid unwanted collisions between your app and other apps on the XMTP network.
+
+```jsx
+import Xmtp from 'xmtp-react-native';
+
+// Start a scoped conversation with ID mydomain.xyz/foo
+let conversation1 = await Xmtp.newConversation("0x3F11b27F323b62B159D2642964fa27C46C841897", "mydomain.xyz/foo")
+
+// Start a scoped conversation with ID mydomain.xyz/bar. And add some metadata
+let conversation2 = await Xmtp.newConversation("0x3F11b27F323b62B159D2642964fa27C46C841897", "mydomain.xyz/bar", ["title": "Bar conversation"])
```
Comment (dmccartney) [inline README.md]: > `["title": "Bar conversation"]`
->
> `{title: "Bar conversation"}`

**Review by dmccartney — APPROVED** [review]

**Review by jhaaaa — APPROVED** [review]
Amazing readme, @nplasterer - thank you!

**Pull request merged**